gtdynamics/optimizer/AugmentedLagrangianOptimizer.cpp

```diff
@@ -47,7 +47,8 @@ void update_parameters(const EqualityConstraints& constraints,
 gtsam::Values AugmentedLagrangianOptimizer::optimize(
     const gtsam::NonlinearFactorGraph& graph,
     const EqualityConstraints& constraints,
-    const gtsam::Values& initial_values) const {
+    const gtsam::Values& initial_values,
+    boost::optional<ConstrainedOptResult*> opt_result) const {
```

Comment:
No, sorry, this should just be `ConstrainedOptResult*`. You can check if it is `!= nullptr` and fill it if not. Finally, if you keep this parameter, just call it `result`. It's optional and i/o by virtue of being a non-const pointer.

Reply:
Done